jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectorViewPanels.kt

```diff
@@ -279,279 +271,247 @@ class EnvironmentDetailsPanel(private val context: CawsSettings, lifetime: Lifet
-                                                elements ?: run {
-                                                    return@proposeModelUpdate
-                                                }
-
-                                                withUiContext {
-                                                    elements.forEach { element -> model.addElement(element) }
-                                                    model.selectedItem = elements.first()
-                                                }
-                                            }
-                                        }
-                                    })
-
-                                    repoUrlField = this
-                                }
-                        }
-
-                        row(message("caws.workspace.details.branch_title")) {
-                            cell(unlinkedBranchCombo)
-                                .bindItem(context::unlinkedRepoBranch.toMutableProperty())
-                                .errorOnApply(message("caws.workspace.details.branch_validation")) {
-                                    !repoUrlField.text.isEmpty() && it.selectedItem == null
-                                }
-                                .columns(COLUMNS_MEDIUM)
                         }
+                        contextHelp(message("caws.one.branch.per.dev.env.comment"))
+                    }.visibleIf(cloneRepoButton.selected)
 
-                        row(message("caws.workspace.details.project.title")) {
-                            textField()
-                                .columns(COLUMNS_MEDIUM)
-                                .comment(message("caws.workspace.details.project.comment"))
-                                .errorOnApply(message("caws.workspace.details.project.required")) { it.text.nullize(true) == null }
-                                .applyToComponent {
-                                    projectField = this
-                                }
-                        }
-                    }
-                }
-
-                // need here to force comboboxes to load
-                if (context.project == null) {
+                    // need here to force comboboxes to load
                     getProjects(client, spaces).forEach { projectCombo.addItem(it) }
-                } else {
-                    projectCombo.addItem(context.project)
-                }
-
-                val propertyGraph = PropertyGraph()
-                val projectProperty = propertyGraph.property(projectCombo.selected())
-                projectCombo.addItemListener {
-                    if (it.stateChange == ItemEvent.SELECTED) {
-                        projectProperty.set(it.item as CawsProject?)
-                    }
-                }
+                    projectCombo.selectedItem = getProjects(client, arrayListOf(CawsSpaceTracker.getInstance().lastSpaceName().toString()))[0]
```

Comment:
make single service call
```kt
getProjects(client, spaces).apply {
    forEach { projectCombo.addItem(it) }
    projectCombo.selectedItem = firstOrNull { it.space == CawsSpaceTracker.getInstance().lastSpaceName() }
}
```

i'm not sure if we actually need to use CawsSpaceTracker here

Reply:
I think from there we can know which space was selected in list of Dev envs, with that we can easily compare

Reply:
i'd rather read this out of the context

Reply:
Will resolve in next related code change.